File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> num_index;

        for(auto i = 0; i < nums.size(); i++){
            auto iter = num_index.find(target-nums[i]);

            if (iter != num_index.end()){
                vector<int> answer = {iter->second, i};
                return answer;
            }

            num_index[nums[i]] = i;
        }

        vector<int> non_found_answer;
        return non_found_answer;
    }
};

File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        existed_nums = {}

        for i in range(len(nums)):
            target_num = target - nums[i] 
            
            if target_num in existed_nums:
                return [existed_nums[target_num], i]

            existed_nums[nums[i]] = i

        return [-1, -1]
        
        